### PR TITLE
add resources to input downloader in the ollama plugin

### PR DESCRIPTION
**Pull request by samhita-alla**
## Why are the changes needed?

To prevent this error from occurring: `failed quota: project-quota: must specify limits.cpu for: input-downloader; limits.memory for: input-downloader` error.

## What changes were proposed in this pull request?

This PR adds resources to the input downloader init container, without which it results in `failed quota: project-quota: must specify limits.cpu for: input-downloader; limits.memory for: input-downloader` error.

## How was this patch tested?
On a prod cluster.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

File: plugins/flytekit-inference/flytekitplugins/inference/sidecar_template.py [hunk 1]
```diff
@@ -138,6 +140,18 @@ def __init__(
                         V1VolumeMount(name="shared-data", mount_path="/shared"),
                         V1VolumeMount(name="tmp", mount_path="/tmp"),
                     ],
+                    resources=V1ResourceRequirements(
+                        requests={
+                            "cpu": 1,
+                            "nvidia.com/gpu": 0,
```
[the remark this answers is not in the document]
Comment (samhita-alla) [reply plugins/flytekit-inference/flytekitplugins/inference/sidecar_template.py:146]: this isn't required. i just tested without adding gpu.

File: plugins/flytekit-inference/flytekitplugins/inference/ollama/serve.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def __init__(
         cpu: int = 1,
         gpu: int = 1,
         mem: str = "15Gi",
+        inputs_mem: str = "500Mi",
```
Comment (thomasjpfan) [inline plugins/flytekit-inference/flytekitplugins/inference/ollama/serve.py:34]: I prefer be more explicit: `download_inputs_mem`.

```suggestion
        download_inputs_mem: str = "500Mi",
```

File: plugins/flytekit-inference/flytekitplugins/inference/sidecar_template.py [hunk 3]
```diff
@@ -138,6 +140,16 @@ def __init__(
                         V1VolumeMount(name="shared-data", mount_path="/shared"),
                         V1VolumeMount(name="tmp", mount_path="/tmp"),
                     ],
+                    resources=V1ResourceRequirements(
+                        requests={
+                            "cpu": 1,
```
Comment (thomasjpfan) [inline plugins/flytekit-inference/flytekitplugins/inference/sidecar_template.py:145]: I can see the need for setting the CPU as well. From my testing, downloads are usually faster if you give it more than one CPU.

**Review by thomasjpfan — COMMENTED** [review]
For my understanding, did this work before without setting the memory?

File: plugins/flytekit-inference/flytekitplugins/inference/sidecar_template.py (hunk 3 above)
Comment (samhita-alla) [reply plugins/flytekit-inference/flytekitplugins/inference/sidecar_template.py]: i made it configurable.

**Comment by samhita-alla** [comment]
> For my understanding, did this work before without setting the memory?

@thomasjpfan it worked on the local flyte cluster. i haven't tested it on the demo cluster earlier because i needed to deploy a backend change to test it.

**Review by eapolinario — APPROVED** [review]

**Pull request merged**